tests/components/azure_data_explorer/conftest.py

```diff
@@ -0,0 +1,85 @@
+"""Test fixtures for ADX."""
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .const import AZURE_DATA_EXPLORER_PATH, BASE_CONFIG, BASIC_OPTIONS, CLIENT_PATH
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@pytest.fixture(autouse=True, name="mock_test_connection")
+def mock_test_connection():
+    """Mock azure data explorer test_connection, used to test the connection."""
+    with patch(f"{CLIENT_PATH}.test_connection") as test_connection:
```

Comment:
You should mock the actual call to the outside API, so mock the actual query, and not the function using the query